### PR TITLE
Refactor app schemas

### PR DESCRIPTION
**Pull request by dannyroberts**
This is all just moving code. Basically

`app_manager/util.py` gets split into

- `app_manager/util.py`
- `app_manager/app_schemas/case_properties.py`
- `app_manager/app_schemas/casedb_schema.py`
- `app_manager/app_schemas/session_schema.py`

and `app_manager/tests/test_util.py` gets split into
- `app_manager/tests/test_util.py`
- `app_manager/app_schemas/tests/test_case_properties.py`
- `app_manager/app_schemas/tests/test_schema.py`

### CODE REVIEW

File: corehq/apps/app_manager/app_schemas/tests/test_schema.py [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+from corehq.apps.app_manager import util as util
+from corehq.apps.app_manager.app_schemas.casedb_schema import get_casedb_schema
+from corehq.apps.app_manager.app_schemas.session_schema import get_session_schema
+from corehq.apps.app_manager.const import USERCASE_TYPE
+from corehq.apps.app_manager.tests.app_factory import AppFactory
+from corehq.util.test_utils import flag_enabled
+
+
+@flag_enabled('USER_PROPERTY_EASY_REFS')
+@patch('corehq.apps.app_manager.app_schemas.casedb_schema.get_case_property_description_dict', MagicMock(return_value={}))
```
Comment (dimagimon) [inline corehq/apps/app_manager/app_schemas/tests/test_schema.py]: E501 line too long (122 > 115 characters)

File: corehq/apps/app_manager/app_schemas/tests/test_schema.py [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+                "id": "parent",
+                "name": "parent (gold-fish)"
+            }
+        ]
+
+        self.assertEqual(casedb_schema['subsets'], expected_casedb_schema_subsets)
+        self.assertEqual(session_schema['structure'], expected_session_schema_structure)
+
+    def test_get_case_sharing_hierarchy(self):
+        with patch('corehq.apps.app_manager.app_schemas.case_properties.get_case_sharing_apps_in_domain') as mock_sharing:
```
Comment (dimagimon) [inline corehq/apps/app_manager/app_schemas/tests/test_schema.py]: E501 line too long (122 > 115 characters)

File: corehq/apps/app_manager/app_schemas/tests/test_schema.py [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+        if case_type:
+            self.factory_2.form_opens_case(form, case_type)
+        if case_updates:
+            assert case_type, 'case_type is required with case_updates'
+            self.factory_2.form_requires_case(
+                form, case_type=case_type, update=case_updates)
+        return form
+
+
+@patch('corehq.apps.app_manager.app_schemas.casedb_schema.get_case_property_description_dict', MagicMock(return_value={}))
```
Comment (dimagimon) [inline corehq/apps/app_manager/app_schemas/tests/test_schema.py]: E501 line too long (122 > 115 characters)

**Comment by dannyroberts** [comment]
@orangejenny this came out of my frustration with working with this code when looking into the slowness associated with `get_casedb_schema` when it touches a ton of apps. This PR does not address that issue; it just separates out the code as a first step, so that it's easier for the next person who comes along and tries to reason with it, figure out its usage footprint and dependencies, etc.

**Review by orangejenny — APPROVED** [review]
This is awesome!

Review was pretty minimal because I'm expecting problems would come up in complilation.

**Comment by dannyroberts** [comment]
Great, I'm just going to give it a basic UI sanity check (often the wiring to the UI isn't tested, and that could easily have broken) and then I'll remove the tag and merge it

**Comment by dannyroberts** [comment]
Basic sanity test passed: opening a form and checking case management in the form builder appears normal. Merging!

**Pull request merged**